torch/_inductor/lowering.py

```diff
@@ -448,20 +448,32 @@ def inner_fn(index):
 def make_foreach_pointwise(pw_fn, allow_alpha=False):
     def inner(*inputs: List[List[TensorBox]], alpha=1):
         # group by device, whether any of the inputs are dynamic, and whether their types match
-        # (proxy for type promotion)
+        # (proxy for type promotion), and whether they are aliased
+        # we count the number of aliases for a given name to determine alias group number
+        # to ensure that two aliases for the same value are never grouped together
         def group_args(arg_pairs):
-            out = defaultdict(list)
+            out: DefaultDict[
+                Tuple[torch.device, bool, int],
+                List[Tuple[int, Tuple[ir.IRNode, ir.IRNode]]],
+            ] = defaultdict(list)
+            alias_counts: DefaultDict[str, int] = defaultdict(int)
             for i, args in enumerate(arg_pairs):
                 use_foreach = not is_dynamic(*args)
                 device = None
+                alias_group = 0
                 for t in args:
                     if isinstance(t, TensorBox):
                         device = t.data.get_device()
-                        break
+                        if isinstance(t.data, ir.BaseView):
+                            unaliased = t.data.unwrap_view()
+                            alias_counts[unaliased.get_name()] += 1
```

Comment:
@jansel Am I guaranteed that the underlying storage will have a name? Curious if you have a more canonical approach to get the originating value for a view that's hashable

Reply:
No, not everything will have a name.  For example, this could be the result of a random number generation where the RNG generation is fused into the foreach kernel.